packages/react-data-grid/src/DataGrid.tsx

```diff
@@ -135,13 +130,10 @@ export interface DataGridProps<R, K extends keyof R> {
   onCellExpand?(options: SubRowOptions<R>): void;
   onRowExpandToggle?(event: RowExpandToggleEvent): void;
 
-  /** InteractionMasksMetaData */
-  /** Deprecated: Function called when grid is updated via a copy/paste. Use onGridRowsUpdated instead*/
-  onCellCopyPaste?(event: CellCopyPasteEvent<R>): void;
-  /** Function called whenever a cell is selected */
-  onCellSelected?(position: Position): void;
-  /** Function called whenever a cell is deselected */
-  onCellDeSelected?(position: Position): void;
+  /** Function called whenever selected cell is changed */
+  onSelectedCellChange?(position: Position): void;
+  /** Function called whenever selected cell range is changed */
+  onSelectedCellRangeChange?(selectedRange: SelectedRange): void;
```

Comment:
It will require some refactoring but I think we can probably use the same event handler. We will have to address a few things
- Currently we attach the mousdown and mouseenter events only when `onCellSelectionChange` is specified. 
- CellRange selection does not work properly when there are frozen columns.

I think it is best to address it separately as the logic can be cleanup